server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -596,6 +601,17 @@ public void test_eq_on_bool_uses_termquery() throws Exception {
         assertThat(query).isExactlyInstanceOf(TermQuery.class);
     }
 
+    @Test
+    public void test_eq_on_alias_uses_termquery() throws Exception {
+        // Testing expression: col as alias = 'foo'
+        AliasSymbol alias = new AliasSymbol("aliased", createReference("name", DataTypes.STRING));
+        var literal = Literal.of("foo");
+        var func = new Function(EqOperator.SIGNATURE, List.of(alias, literal), DataTypes.BOOLEAN);
+        Query query = queryTester.toQuery(func);
+        assertThat(query).isNotExactlyInstanceOf(GenericFunctionQuery.class);
+        assertThat(query.toString()).doesNotContainIgnoringCase("aliased");
```

Comment:
Any reason not to check for the term query?

Reply:
Thank you, this is recovered from https://github.com/crate/crate/commit/7f763496099d2daa51fa1734da2fada32bad265f and forgot to update the title. Fixing the title as `test_function_arguments_are_unaliased`.